src/pip/_internal/metadata/importlib/_dists.py

```diff
@@ -100,8 +101,8 @@ class Distribution(BaseDistribution):
     def __init__(
         self,
         dist: importlib.metadata.Distribution,
-        info_location: Optional[BasePath],
-        installed_location: Optional[BasePath],
+        info_location: Optional[Union[BasePath, pathlib.PurePosixPath]],
```

Comment:
PurePosixPath doesn't have exists and I didn't find more elegant way to solve it

Reply:
https://github.com/TheQuist1994